ProcessMaker/processmaker
# Implement Backend and Frontend Functionality for Guided Templates

This PR introduces backend and frontend functionality for Guided Templates. Key features include:

**Backend:**

1. Implementation of the artisan command `php artisan processmaker:sync-wizard-templates` to sync guided templates from the wizard-templates repository.
2. Migrations to update the schema of the `wizard_templates` table to accommodate the newly structured guided templates.
3. Migration for the `processes`, `scripts`, and `screens` tables to include the `asset_type` column, ensuring guided template assets are hidden from the UI consistently.
4. Implementation of an API to trigger the start event of the helper process when configuring a guided template.

**Frontend:**

1. Implementation of the updated wizard template database structure to render UI elements such as icons, background images, slider images, etc.
2. Rendering of helper process tasks in a modal when selecting the 'Use Now' button.
3. Implementation of search functionality for guided templates.

## How to Test

1. Ensure you have local access to a development environment.
2. Add the following variables to your .env file:
```
WIZARD_TEMPLATE_REPO=wizard-templates
WIZARD_TEMPLATE_BRANCH=task/FOUR-12701
WIZARD_TEMPLATE_CATEGORIES=all
```

3. Run php artisan optimize:clear.
4. Run php artisan processmaker:sync-wizard-templates.
5. Upon completion of the command, navigate to 'Processes -> Guided Templates'.
6. Ensure the example guided template is populated.
7. Verify that selecting the guided template opens the template detail modal.
8. Check that selecting the 'Use Now' button renders the modal with helper process tasks to configure the guided template.
9. Ensure that upon completion of the process, the modal is closed.

## Related Tickets & Packages
- [Collections PR](https://github.com/ProcessMaker/package-collections/pull/371)
- [Wizard Template Repo PR](https://github.com/ProcessMaker/wizard-templates/pull/11)

## Resolves Tickets:
-  [FOUR-12701](https://processmaker.atlassian.net/browse/FOUR-12701)
-  [FOUR-12648](https://processmaker.atlassian.net/browse/FOUR-12648)
-  [FOUR-12178](https://processmaker.atlassian.net/browse/FOUR-12178)
-  [FOUR-12179](https://processmaker.atlassian.net/browse/FOUR-12179)
-  [FOUR-11911](https://processmaker.atlassian.net/browse/FOUR-11911)

ci:package-collections:task/FOUR-12701
ci:next

## Code Review Checklist


[FOUR-12701]: https://processmaker.atlassian.net/browse/FOUR-12701?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
